api/python/quilt3/packages.py

```diff
@@ -1101,7 +1101,19 @@ def push(self, name, registry=None, dest=None, message=None):
 
         self._fix_sha256()
         pkg = self._materialize(dest)
+
+        def physical_key_is_temp_file(pk):
+            if not file_is_local(pk):
+                return False
+            return pathlib.Path(parse_file_url(urlparse(pk))).parent == APP_DIR_TEMPFILE_DIR
+        temp_file_logical_keys = [lk for lk, entry in self.walk() if physical_key_is_temp_file(entry.physical_keys[0])]
```

Comment:
This logic is now duplicated here and in `_delete_temporary_files`... Maybe do it all at once - find temp files, delete them, and fix the logical keys?

Reply:
Good point. Updated it so we only do that work once